crates/notary/server/src/server.rs

```diff
@@ -210,6 +210,15 @@ pub async fn run_server(config: &NotaryServerProperties) -> Result<(), NotarySer
                     }
                     Err(err) => {
                         error!("{}", NotaryServerError::Connection(err.to_string()));
+                        if let Ok(error) = err.downcast::<rustls::Error>() {
+                            if error
+                                == rustls::Error::InvalidMessage(
+                                    rustls::InvalidMessage::InvalidContentType,
+                                )
```

Comment:
```
  if let Some(rustls::Error::InvalidMessage(rustls::InvalidMessage::InvalidContentType)) =
       err.get_ref().and_then(|inner| inner.downcast_ref::<rustls::Error>())
```

this could be neater